workflow/util/util.go

```diff
@@ -839,6 +852,14 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 	return newWF, podsToDelete, nil
 }
 
+func resetNode(node wfv1.NodeStatus) wfv1.NodeStatus {
+	node.Phase = wfv1.NodeRunning
+	node.Message = ""
+	node.StartedAt = metav1.Time{Time: time.Now().UTC()}
+	node.FinishedAt = metav1.Time{}
+	return node
+}
+
```

Comment:
I really like that this was broken out into its own function.
One thing I was looking at was NodeStatus being passed in by value rather than pointer (and thus the copy being made). I guess it does avoid having to check for nil, and it's also consistent with the function below. Plus, this won't be called that often, right? What do you think?